commit/merkleroot/outcome.go

```diff
@@ -316,7 +328,8 @@ func checkForReportTransmission(
 		}
 	}
 
-	if offRampUpdated {
+	// All pending sources have been updated, we can move to the next state.
+	if len(pendingSources) == 0 {
```

Comment:
@dimkouv could you double check the logic implemented above?

1. Initialize a set based on all chain selectors from `RootsToReport`. (this may already be broken if there are multiple roots for a single source chain selector.
2. While looping over previous and new seqNums, delete source selectors from the set. The idea being we'll have an empty set at the end if all source selectors have been updated.
3. If there is only 1 report, the logic is short-circuited similar to before.
4. At the end, if there are 0 remaining sources, then transmission must have completed.

Reply:
> Added two comments above. We'll also need a follow-up to keep making progress with the source chains that got their reports transmitted and not keep them blocked due to other source chains.

There's a `ReportTransmissionCheckAttempts` limit, I think that should prevent good transmissions from being completely blocked by a bad one. But if a bad report keeps recurring every plugin cycle would be slowed down at the checkReportTransmission phase.

Do you have a solution in mind? I'm not sure what to put in a ticket for this case.